src/util/get-type-hint-from-type.ts

```diff
@@ -11,7 +18,11 @@ export function getTypeHintFromType(type: string | Type | SimpleType | undefined
 	if (type == null) return undefined;
 	if (typeof type === "string") return type;
 
-	const typeHint = toTypeString(type, checker);
+	let simpleType: SimpleType = isSimpleType(type) ? type : toSimpleType(type, checker);
+	if (isStringLiteralsUnionAlias(simpleType)) {
+		simpleType = simpleType.target;
+	}
+	const typeHint = simpleTypeToString(simpleType);
```

Comment:
If possible, I would prefer not converting all types to SimpleType (due to overhead). You couldn't know, but actually the `toTypeString` function didn't convert native Typescript types to SimpleType, - in that case it just used `checker.typeToString` behind the scenes. I've found that if we give the `TypeFormatFlags.InTypeAlias` flag to `checker.typeToString`, we get the intended output using the native Typescript function. Therefore, I think we should go with the `isStringLiteralsUnionAlias`-check when dealing with a `SimpleType` and `TypeFormatFlags.InTypeAlias` otherwise. Something like: 

```typescript
let typeHint: string;
if (isSimpleType(type)) {
  if (isStringLiteralsUnionAlias(type)) {
    type = type.target;
  }
	
  typeHint = toTypeString(type, checker);
} else {
  typeHint = checker.typeToString(type, undefined, TypeFormatFlags.InTypeAlias);
}

Reply:
Perfect, I think it totally makes sense, thank you for the tips!
Now the thing is that if the type is `SimpleType` and is not only a type alias of only strings is not converting it. It will leave the alias name, but if the type is a native `Type` it will always print the union. (Not sure if I am explaining myself correctly 😅).
So I think it makes sense to have the same behavior with `SimpleType` and the code will be also simpler.